### PR TITLE
Generate staging DB with fake data

### PR DESCRIPTION
**Pull request by pylipp**
- added a couple of CRUD functions
- for details of the fake data generation, see the comments in `data_faking.py`
- `init.sql` is updated such that it contains only the data generated by the Minimal, DemoUsers, and Cypress PHP scripts in https://github.com/boxwise/dropapp/tree/5ae4408163e963f479c6ea33788c7993c8f00be5/db/seeds 

Functional test:
- `docker compose up --build webapp` then
- `curl 'http://localhost:5005/cron/reseed-db' -H 'x-appengine-cron: true'`

Needs clarification:
- how realistic do datetime fields need to be? currently they reflect the time the script runs

### CODE REVIEW

File: back/boxtribute_server/cron/data_faking.py [hunk 1]
```diff
@@ -0,0 +1,838 @@
...
+from ..models.definitions.base import Base
+from ..models.definitions.box import Box
+from ..models.definitions.product import Product
+from ..models.definitions.size import Size
+from ..models.utils import convert_ids
+
+NR_BASES = 4
+NR_OF_CREATED_TAGS_PER_BASE = 20
+NR_OF_DELETED_TAGS_PER_BASE = round(NR_OF_CREATED_TAGS_PER_BASE / 10)
+LOCATION_NAMES = ("Stockroom", "WH", "FreeShop", "LOST", "SCRAP", "Unused WH")
```
Comment (HaGuesto) [inline back/boxtribute_server/cron/data_faking.py:55]: no LOST and SCRAP locations pls

Comment (HaGuesto) [reply back/boxtribute_server/cron/data_faking.py:55]: and change to WH2 and another donated location

File: back/boxtribute_server/cron/data_faking.py [hunk 2]
```diff
@@ -0,0 +1,838 @@
...
+NR_OF_CREATED_LOCATIONS_PER_BASE = len(LOCATION_NAMES)
+NR_OF_ADULTS_PER_BASE = 100
```
Comment (HaGuesto) [inline back/boxtribute_server/cron/data_faking.py:57]: pls create only one base with 300 beneficiaries (load testing) and the rest with less than half the amount.

File: back/boxtribute_server/cron/data_faking.py [hunk 3]
```diff
@@ -0,0 +1,838 @@
...
+NR_OF_CHILDREN_PER_BASE = 200
+NR_OF_BENEFICIARIES_PER_BASE = NR_OF_ADULTS_PER_BASE + NR_OF_CHILDREN_PER_BASE
+NR_OF_BOXES_PER_BASE = 100
```
Comment (HaGuesto) [inline back/boxtribute_server/cron/data_faking.py:60]: pls also create one base with 300 or 500 boxes for load testing

File: back/boxtribute_server/business_logic/warehouse/location/crud.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    location = Location.get_by_id(id)
+
+    if name is not None:
+        location.name = name
+    if description is not None:
+        location.description = description
+    if box_state is not None:
+        location.box_state = box_state
+        location.is_donated = box_state == BoxState.Donated
+        location.is_lost = box_state == BoxState.Lost
```
Comment (HaGuesto) [inline back/boxtribute_server/business_logic/warehouse/location/crud.py:75]: take out lost and scrap

**Review by HaGuesto — APPROVED** [review]
# code review passed

File: back/boxtribute_server/business_logic/warehouse/location/crud.py (hunk 4 above)
Comment (pylipp) [reply back/boxtribute_server/business_logic/warehouse/location/crud.py:75]: Does it make sense to keep it for consistency until the `is_lost` and `is_scrap` fields are removed from the locations table?

**Comment by pylipp** [comment]
[2.0 Data faking for test DB seed](https://trello.com/c/oKpbiC3M/1402-20-data-faking-for-test-db-seed)

**Pull request merged**